junit-jupiter-api/src/testFixtures/java/org/junit/jupiter/api/ClasspathRunner.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+/**
+ * Utility class for running code with a custom classpath.
+ *
+ * @since 5.13
+ */
+public class ClasspathRunner {
```

Comment:
I'm not sure the class name really captures its essence... let me know if you have a better proposal!

Reply:
Maybe sth. like `ExtraClasspathRootExecutor`?

Reply:
Maybe the method should also be more specific.

What do you think about `TemporaryClasspathExecutor` and `withExtraClasspath` / `withAdditionalClasspath`?

Reply:
`TemporaryClasspathExecutor.withAdditionalClasspathRoot` sounds good to me! :+1: